lib/std/heap/arena_allocator.zig

```diff
@@ -3,8 +3,9 @@ const assert = std.debug.assert;
 const mem = std.mem;
 const Allocator = std.mem.Allocator;
 
-/// This allocator takes an existing allocator, wraps it, and provides an interface
-/// where you can allocate without freeing, and then free it all together.
+/// This allocator takes an existing allocator, wraps it, and provides an interface where
+/// you can allocate and then free it all together. You can only free an individual item
+/// if it was the most recent allocation, otherwise calls to free do nothing.
```

Comment:
Imprecise. "You can only free an individual item" is ambiguous. It is clarified later with "calls to free do nothing" but the docs should avoid creating ambiguity in the first place.

Reply:
Thanks, I've updated the wording to try and avoid the ambiguity.